drivers/unix/os_unix.cpp

```diff
@@ -375,19 +375,61 @@ void OS_Unix::delay_usec(uint32_t p_usec) const {
 	}
 }
 
+void OS_Unix::precise_delay_usec(uint32_t p_usec) const {
+	thread_local double estimate = 5000.0;
+	thread_local double mean = 5000.0;
+	thread_local double m2 = 0.0;
+	thread_local uint64_t count = 1;
+
+	uint64_t t0 = get_ticks_usec();
+	uint64_t target_time = t0 + p_usec;
+	double usec_d = p_usec;
+
+	// Perform coarse sleep while the remaining time exceeds the estimate.
+	while (usec_d > estimate) {
+		uint64_t sleep_start = get_ticks_usec();
+		delay_usec(1000);
+		uint64_t sleep_end = get_ticks_usec();
+
+		// Compute observed time and limit to 1 second.
+		double observed = sleep_end - sleep_start;
+		usec_d -= observed;
+		observed = MIN(observed, 1000000.0);
+
+		// Update statistical estimates for mean and standard deviation.
+		if (unlikely(count > 1000000)) {
+			mean = (mean + observed) / 2;
+			m2 = 0.0;
+			count = 1;
+		}
+
+		++count;
+		double delta = observed - mean;
+		mean += delta / count;
+		m2 += delta * (observed - mean);
+		double stddev = Math::sqrt(m2 / (count - 1));
+		estimate = mean + stddev;
+	}
```

Comment:
Do we need all these logic if we need the sleep is always going to be sub 1 second?

Reply:
Or... given the current implementation runs based on estimates, why does the code that runs the estimates need to be platform-dependent?

Reply:
So, given this is going to be hardware dependent, should we instead use the estimate logic everywhere (and just move it inside `add_frame_delay` instead of creating a dedicated function in each platform)?

Reply:
Are we sure that on windows we don't need the estimation? I.e. do we know for sure that on windows there's not going to be differences depending on the hardware we are running on?

Reply:
Well, according to [this](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleep#remarks) and [this](https://learn.microsoft.com/en-us/windows/win32/api/timeapi/nf-timeapi-timebeginperiod), `timeBeginPeriod(1)` does not guarantee that the sleep will indeed result in that resolution (should we check `timeGetDevCaps` too maybe?), the same way Unix [`clock_getres`](https://pubs.opengroup.org/onlinepubs/000095399/functions/clock_getres.html) will tell you the resolution of the MONOTONIC clock, but also won't guarantee that `sleep` time (i.e. that would be a lower bound).

Reply:
Please stop marking review comments as resolved.
It's really annoying.
It's the reviewer responsibility to do so once the answer provided are deemed satisfying.

Reply:
How would you prefer feedback to be given instead? Please try to communicate your issues it helps to know what you are having an issue with

Reply:
(thank you, my comment accidentally deleted when I was editing it, clumsy mobile interface)